src/frontend/web_application/src/store/modules/message.js

```diff
@@ -87,24 +107,67 @@ export function hasMore(state) {
   return state.total > state.messages.length;
 }
 
+function messagesByIdReducer(state = {}, action = {}) {
+  switch (action.type) {
+    case REQUEST_MESSAGES_SUCCESS:
+      return action.payload.data.messages.reduce((previousState, message) => ({
+        ...previousState,
+        [message.message_id]: message,
+      }), state);
+    case SYNC_MESSAGE:
+      return {
+        ...state,
+        [action.payload.message.message_id]: action.payload.message,
+      };
+    default:
+      return state;
+  }
+}
+
+function messageIdsByDiscussionIdReducer(state = {}, action = {}) {
+  if (action.type !== REQUEST_MESSAGES_SUCCESS) {
+    return state;
+  }
+
+  const applyMessageId = (messagesState = [], message) => {
+    const nextState = [...messagesState];
+    if (nextState.indexOf(message.message_id) === -1) {
+      nextState.push(message.message_id);
+    }
+
+    return nextState;
+  };
+
+  return action.payload.data.messages.reduce((acc, message) => {
+    const { discussion_id: discussionId } = message;
+
+    return {
+      ...acc,
+      [discussionId]: applyMessageId(acc[discussionId], message),
+    };
+  }, { ...state });
```

Comment:
Unnecessarily creating a new object, and merging the content of another into it, adding junk code after babel-transpile ...

It's absolutely safe to just use `state` here